feat(note): migrate the note component (#400)
Vonage/vivid-3: libs/components/src/lib/note/README.md

# note

`vwc-note` component is designated to layout connotated notification content.

```js
<script type="module">import '@vonage/vivid/note';</script>
```

```html preview
<vwc-note connotation="success"
          icon="check-circle"
          header="Pascal's theological argument">
    Pascal argues that a rational person should live as though God exists and seek to believe in God. If God does not actually exist, such a person will have only a finite loss (some pleasures, luxury, etc.), whereas if God does exist, he stands to receive infinite gains (as represented by eternity in Heaven) and avoid infinite losses (eternity in Hell).
</vwc-note>
```

## Properties

> ### header

Reviewer: ```suggestion
### heading
```
the text is not really the note header but more its heading. Like in Card:
https://vivid-6066n7078jtg.deno.dev/components/card#heading
I think we need to change it and its class name :)

> Add the `header` attribute in order to set a header text.

Reviewer 2: ```suggestion
Add the `heading` attribute in order to set a header text.
```

- Type: `string`

> - Default: `null`

Reviewer: ```suggestion
- Default: `undefined`
```

```html preview
<vwc-note header="Header Text"></vwc-note>
```

### icon

> Add the `icon` attribute in order to set an icon in the note.

Reviewer: ```suggestion
Add a icon='icon-name' attribute  in order to set an icon in the note.
```

Reviewer 2: we have so many variants of the same member. we should consolidate to one. or at least not produce more variants.

having said that, here's a new variant :D

```suggestion
Use the `icon` attribute to set a decorative icon.
Note that if not set, icon will default to the connotation-associated icon type.
```

- Type: `string`

> - Default: `null`

Reviewer: ```suggestion
- Default: `undefined`
```

```html preview
<vwc-note icon="home"></vwc-note>
```

### connotation
Set the `connotation` attribute to change the note's connotation.
It accepts a subset of predefined values.

- Type: `'success'` | `'alert'` | `'warning'` | `'info'` | `'announcement'`
- Default: `'announcement'`

```html preview

> <vwc-note connotation="alert"></vwc-note>
> <vwc-note connotation="success"></vwc-note>
> <vwc-note connotation="warning"></vwc-note>
> <vwc-note connotation="info"></vwc-note>
> <vwc-note connotation="announcement"></vwc-note>

Reviewer: the preview looks strange with no text inside - I think we can add:
```suggestion
<vwc-note connotation="alert" header="alert note"></vwc-note>
<vwc-note connotation="success" header="success note"></vwc-note>
<vwc-note connotation="warning" header="warning note"></vwc-note>
<vwc-note connotation="info" header="info note"></vwc-note>
<vwc-note connotation="announcement" header="announcement note"></vwc-note>
```
